components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java

```diff
@@ -2001,8 +2001,8 @@ private void sendEmailNotification(API api) throws APIManagementException {
         }
     }
 
-    private void validateAndSetTransports(API api) throws APIManagementException {
-        String transports = api.getTransports();
+    private void validateAndSetTransports(API api, APIProduct apiProduct) throws APIManagementException {
```

Comment:
Its not a good design to have a single method like this to handle both API and APIProduct scenario, this leads to lots of null checks that can be avoided and it means the function user has to purposefully pass null as one of the parameters.

Lets define an overloaded function for APIProduct _void validateAndSetTransports(APIProduct apProduct)_ to handle this.